modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -526,14 +533,17 @@ def cost_estimate(self, circuit: Circuit, n_shots: int) -> float:
                 "Circuit does not satisfy predicates of backend."
                 + " Try running `backend.get_compiled_circuit` first"
             )
-        gate_counts: Dict[OpType, int] = {
-            g_type: circuit.n_gates_of_type(g_type) for g_type in _GATE_SET
-        }
-        n_1q = gate_counts[OpType.PhasedX]
-        n_m = circuit.n_qubits + gate_counts[OpType.Measure] + gate_counts[OpType.Reset]
-        n_2q = gate_counts[OpType.ZZMax]
-        cost: float = 5 + (n_1q + 10 * n_2q + 5 * n_m) * n_shots / 5000
-        return cost
+        if self._device_name.endswith("SC"):
+            backend = self
+        elif self._device_name.endswith("E"):
+            backend = QuantinuumBackend(self._device_name[:-1] + "SC")
+        else:
+            backend = QuantinuumBackend(self._device_name + "SC")
```

Comment:
Is there any error handling taking place in case this device name is not valid?

Reply:
have added some error handling